controllers/horizon_controller.go

```diff
@@ -717,7 +718,8 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrlResult, nil
 	}
 	instance.Status.ReadyCount = depl.GetDeployment().Status.ReadyReplicas
-	if instance.Status.ReadyCount > 0 {
+	if instance.Status.ReadyCount > 0 &&
```

Comment:
we need to fix this to compare with the Spec.replicas instead of > 0 otherwise the horizon can be ready before all the replicas are updated to the new image

Reply:
I think this code is there because of the definition of `DeploymentReadyCondition`: I think we should change it to something like: `instance.Status.ReadyCount == *instance.Spec.Replicas`, which is the same thing we have in other operators.